query_modules/schema.cpp

```diff
@@ -92,31 +92,65 @@ void Schema::ProcessPropertiesRel(mgp::Record &record, const std::string_view &t
   record.Insert(std::string(kReturnMandatory).c_str(), mandatory);
 }
 
+// Custom equality operator for vector of strings
+// We need this because Label1:Label2 and Label2:Label1 are the same
+struct VectorComparator {
+  bool operator()(const std::vector<std::string> &lhs, const std::vector<std::string> &rhs) const {
+    // Sort and compare the vectors of strings
+    std::vector<std::string> sortedLhs = lhs;
+    std::vector<std::string> sortedRhs = rhs;
+    std::sort(sortedLhs.begin(), sortedLhs.end());
+    std::sort(sortedRhs.begin(), sortedRhs.end());
+
+    return sortedLhs < sortedRhs;
+  }
+};
+
 void Schema::NodeTypeProperties(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
   mgp::MemoryDispatcherGuard guard{memory};
-  ;
   const auto record_factory = mgp::RecordFactory(result);
   try {
-    const mgp::Graph graph = mgp::Graph(memgraph_graph);
-    for (auto node : graph.Nodes()) {
+    std::map<std::vector<std::string>, std::tuple<std::set<std::string>, std::string, mgp::List>, VectorComparator>
+        node_types;  // map of node types, key is vector of labels, value is tuple of set of properties, type of node
```

Comment:
Why is `labels` key in the form of labels_vector, and value in form of labels, this seems like duplicated info?